src/sdam/topology.ts

```diff
@@ -881,13 +931,33 @@ function processWaitQueue(topology: Topology) {
         : serverDescriptions;
     } catch (e) {
       waitQueueMember.timeoutController.clear();
-
+      topology.client.mongoLogger.debug(
+        MongoLoggableComponent.SERVER_SELECTION,
+        new ServerSelectionFailedEvent(
+          waitQueueMember.serverSelector,
+          topology.description,
+          e,
+          waitQueueMember?.operationName
+        )
+      );
       waitQueueMember.callback(e);
       continue;
     }
 
-    let selectedServer;
+    let selectedServer: Server | undefined;
     if (selectedDescriptions.length === 0) {
+      if (!waitQueueMember.waitingLogged) {
+        topology.client.mongoLogger.info(
+          MongoLoggableComponent.SERVER_SELECTION,
+          new WaitingForSuitableServerEvent(
+            waitQueueMember.serverSelector,
+            topology.description,
+            waitQueueMember.timeoutController.getRemainingTimeMS(),
+            waitQueueMember?.operationName
```

Comment:
If we always can access waitQueueMember.timeoutController does it mean that we don't need to check if waitQueueMember exists to access operationName? So basically operationName will always be present?

Reply:
I don't think we need optional chaining here, I've removed it